modROS.lua

```diff
@@ -505,32 +506,38 @@ function ModROS:rosPubMsg(flag)
             return
         end
 
+        -- raycastNode initialization
         local vehicle = g_currentMission.controlledVehicle
-        self.instance_veh = VehicleCamera:new(vehicle, ModROS)
-        local xml_path = self.instance_veh.vehicle.configFileName
-        local xmlFile = loadXMLFile("vehicle", xml_path)
-        -- index 0 is outdoor camera; index 1 is indoor camera
-        -- local cameraKey = string.format("vehicle.enterable.cameras.camera(%d)", 0)
-
-        --  get the cameraRaycast node 2(on top of ) which is 0 index .raycastNode(0)
-        --  get the cameraRaycast node 3 (in the rear) which is 1 index .raycastNode(1)
-
-        local cameraKey = string.format("vehicle.enterable.cameras.camera(%d).raycastNode(0)", 0)
-        XMLUtil.checkDeprecatedXMLElements(xmlFile, xml_path, cameraKey .. "#index", "#node") -- FS17 to FS19
-        local camIndexStr = getXMLString(xmlFile, cameraKey .. "#node")
-        self.instance_veh.cameraNode =
-            I3DUtil.indexToObject(
-            self.instance_veh.vehicle.components,
-            camIndexStr,
-            self.instance_veh.vehicle.i3dMappings
-        )
-        if self.instance_veh.cameraNode == nil then
-            print("nil camera")
-        -- else
-        --     print(instance_veh.cameraNode)
+        -- if the player is not in the vehicle, print error and return
+        if not vehicle then
```

Comment:
This will avoid the `nil` related error, but should `self.rosPubMsg` also not be reset to `false` in this case?

Otherwise I believe `ModROS:update(..)` will try to still publish data, right?

Reply:
And in that case we should probably also guard the opening of the pipe file, otherwise that will be attempted multiple times.

Reply:
But 1cb72b506df81b10f96cde28c193726d97722587 also avoids publishing data if the player is not in a vehicle

Reply:
But why even continue with `rosPubMsg(..)` if it's clear the player is not in a vehicle?

By adding more conditionals everywhere it looks like we're just complicating things which could be kept simple.

Reply:
because initially I wanted to control the inactive vehicles as well.

Reply:
So are you planning to change anything, or should we merge this version?

Reply:
yes, we should just merge this version for now. Even though it's not a proper fix, it does solve the issue I believe.